File: compare-the-triplets/compare-the-triplets.clj
(defn compareTriplets [a b]
  (reduce
   (fn [ret index]
     (let [[aPoints bPoints] ret
           aItem (nth a index)
           bItem (nth b index)
           isAPoint (> aItem bItem)
           isBPoint (< aItem bItem)]
       (cond
         isAPoint (vector (inc aPoints) bPoints)
         isBPoint (vector aPoints (inc bPoints))
         :else ret)))
   (vector 0 0)
   (range (count a))))

File: compare-the-triplets/compare-the-triplets.go
package main

func compareTriplets(a []int32, b []int32) []int32 {
	var aPoints int32 = 0
	var bPoints int32 = 0

	for i := 0; i < len(a) && i < len(b); i++ {
		aItem := a[i]
		bItem := b[i]
		if aItem > bItem {
			aPoints++
		} else if bItem > aItem {
			bPoints++
		}
	}

	return []int32{aPoints, bPoints}
}
